### PR TITLE
Merged Valheim and Valheim Plus templates.

### PR DESCRIPTION
**Pull request by gkuba**
I decided to merge the two Valheim templates into one.

Removed:

- valheimplus.kvp
- Server category in Configuration

Added:

- Changed the saves folder name from Data to Saves for clarity to the user. 
- BepInEx and other needed EnvironmentVariables
- Valheim General Settings category in Configuration
- Valheim Plus Settings category in Configuration
- Moved the Steam Update and Valheim Plus downloads to the valheimupdates.json 
- A Install Valheim Plus toggle under Configuration -> Valheim Plus Settings
- Info box about configuring Valheim Plus until I can get the settings from the .cfg file added in a good way. 
- Added update config to move old Data dir to Saves

Fixed:

- Fixed the Regex for player joined and leaves server to use the steamid and username variable as a workaround to userid not working as intended.

### CODE REVIEW

**Comment by IceOfWraith** [comment]
I haven't reviewed this yet, but excited to see how it behaves! The one thing that we'll need to discuss that I see just from your summary is the saves folder. Changing that may cause issues for people with existing game saves that update their template. It may not be the case though, Have you tried that by chance?

**Comment by IceOfWraith** [comment]
After some quick thought... Perhaps a rename command would solve that. Just rename the Data directory to Saves as an update stage on each OS.

**Comment by gkuba** [comment]
I could add that to solve any potential issues with the directory name change.

**Comment by IceOfWraith** [comment]
Never rush this stuff. Nobody is gonna be screaming that you get it done. Rest and come back to it. :)

**Comment by Greelan** [comment]
Note that on some distros `cp` is aliased so that it interactively prompts for overwriting. The UT99 kvp is an example of how to work around that.

**Comment by gkuba** [comment]
Yeah that was a copy pasta mistake. Its fixed now to use mv for less confusion after the conversion.

**Comment by gkuba** [comment]
Ok I think that is all good to go for now. Tests on both Linux and Windows worked I even tested with an already existing Data folder. I'm going to leave this be for the night and get some sleep. Thanks for all the help guys!

**Comment by gkuba** [comment]
Added the updated user Joined / Leaves server regex for issue #242

**Comment by gkuba** [comment]
I haven't tested every config option for the Valheim Plus settings. I have tested the saves folder change as well as the BepInEx install without Valheim Plus. Both ware working on Linux and Windows.

**Comment by Greelan** [comment]
Marked as draft pending testing of the config options

File: valheimconfig.json [hunk 1]
```diff
@@ -94,5 +87,4644 @@
...
+    "InputType":"text",
+    "IsFlagArgument":false,
+    "ParamFieldName":"AdvancedEditingMode.copyObjectRotation",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"Keypad7",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Apply Copied Object Rotation Hotkey",
+    "Category":"Valheim Plus - Server Settings",
+    "Description":"; Apply the copied object rotation of the currently selected object in Advanced Editing Mode. Default Keypad8",
```
Comment (IceOfWraith) [inline valheimconfig.json:4074]: `; ` needs removed

File: valheimconfig.json [hunk 2]
```diff
@@ -94,5 +87,4644 @@
...
+    "InputType":"text",
+    "IsFlagArgument":false,
+    "ParamFieldName":"AdvancedEditingMode.confirmPlacementOfAdvancedEditingMode",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"KeypadEnter",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Copy Object Rotation Hotkey",
+    "Category":"Valheim Plus - Server Settings",
+    "Description":"; Copy the object rotation of the currently selected object in Advanced Editing Mode. Default Keypad7",
```
Comment (IceOfWraith) [inline valheimconfig.json:4062]: `; ` needs removed

File: valheimconfig.json [hunk 3]
```diff
@@ -94,5 +87,4644 @@
...
+    "Keywords":"Advanced,Building,Mode,Hotkey,Decrease,Scroll,Speed",
+    "FieldName":"AdvancedBuildingMode.decreaseScrollSpeed",
+    "InputType":"text",
+    "IsFlagArgument":false,
+    "ParamFieldName":"AdvancedBuildingMode.decreaseScrollSpeed",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"KeypadMinus",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Camera Settings Enable",
```
Comment (IceOfWraith) [inline valheimconfig.json:3997]: Duplicate DisplayName

File: valheimconfig.json [hunk 4]
```diff
@@ -94,5 +87,4644 @@
...
+    "Keywords":"Advanced,Editing,Mode,Hotkey",
+    "FieldName":"AdvancedEditingMode.DecreaseScrollSpeed",
+    "InputType":"text",
+    "IsFlagArgument":false,
+    "ParamFieldName":"AdvancedEditingMode.DecreaseScrollSpeed",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"KeypadMinus",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Camera Settings Enable",
```
Comment (IceOfWraith) [inline valheimconfig.json:4108]: Duplicate DisplayName

File: valheimconfig.json [hunk 5]
```diff
@@ -94,5 +87,4644 @@
...
+    "Keywords":"Grid,Alignment,Mode,Hotkey",
+    "FieldName":"GridAlignment.changeDefaultAlignment",
+    "InputType":"text",
+    "IsFlagArgument":false,
+    "ParamFieldName":"GridAlignment.changeDefaultAlignment",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"F6",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Camera Settings Enable",
```
Comment (IceOfWraith) [inline valheimconfig.json:4159]: Duplicate DisplayName

File: valheimconfig.json [hunk 6]
```diff
@@ -94,5 +87,4644 @@
...
+    "Keywords":"Advanced,Building,Mode,Hotkey,Rotation,Object,Copy",
+    "FieldName":"AdvancedBuildingMode.copyObjectRotation",
+    "InputType":"text",
+    "IsFlagArgument":false,
+    "ParamFieldName":"AdvancedBuildingMode.copyObjectRotation",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"Keypad7",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Copy Object Rotation Hotkey",
```
Comment (IceOfWraith) [inline valheimconfig.json:3961]: Duplicate DisplayName

File: valheimconfig.json [hunk 7]
```diff
@@ -94,5 +87,4644 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Fermenter.enable",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "false":"false",
+      "true":"true"
+    }
+  },{
+    "DisplayName":"Frementer Durration",
```
Comment (IceOfWraith) [inline valheimconfig.json:677]: `Duration`

File: valheimconfig.json [hunk 8]
```diff
@@ -94,5 +87,4644 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"FireSource.ignorePrivateAreaCheck",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"true",
+    "EnumValues":{
+      "false":"false",
+      "true":"true"
+    }
+  },{
+    "DisplayName":"Frementer Chest Detection Range",
```
Comment (IceOfWraith) [inline valheimconfig.json:851]: Duplicate DisplayName

File: valheimconfig.json [hunk 9]
```diff
@@ -94,5 +87,4644 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Furnace.autoFuel",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "false":"false",
+      "true":"true"
+    }
+  },{
+    "DisplayName":"Furnace Ignore Private Area Check",
```
Comment (IceOfWraith) [inline valheimconfig.json:961]: Duplicated setting. Seems to be identical to the other with the same name.

File: valheimconfig.json [hunk 10]
```diff
@@ -94,5 +87,4644 @@
...
+    "FieldName":"Kiln.stopAutoFuelThreshold",
+    "InputType":"number",
+    "IsFlagArgument":false,
+    "ParamFieldName":"Kiln.stopAutoFuelThreshold",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{},
+    "Suffix": "coal"
+  },{
+    "DisplayName":"Ignore Private Area Check",
```
Comment (IceOfWraith) [inline valheimconfig.json:1615]: Duplicate DisplayName

File: valheimconfig.json [hunk 11]
```diff
@@ -94,5 +87,4644 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"CraftFromChest.checkFromWorkbench",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"true",
+    "EnumValues":{
+      "false":"false",
+      "true":"true"
+    }
+  },{
+    "DisplayName":"Ignore Private Area Check",
```
Comment (IceOfWraith) [inline valheimconfig.json:446]: Duplicate DisplayName

File: valheimconfig.json [hunk 12]
```diff
@@ -94,5 +87,4644 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Kiln.autoDeposit",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "false":"false",
+      "true":"true"
+    }
+  },{
+    "DisplayName":"Kiln Auto Deposit",
```
Comment (IceOfWraith) [inline valheimconfig.json:1587]: Duplicate DisplayName

File: valheimconfig.json [hunk 13]
```diff
@@ -94,5 +87,4644 @@
...
+    "Keywords":"Player,Projectile,Min,Accuracy",
+    "FieldName":"PlayerProjectile.playerMinChargeAccuracyMultiplier",
+    "InputType":"number",
+    "IsFlagArgument":false,
+    "ParamFieldName":"MonsterProjectile.playerMinChargeAccuracyMultiplier",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Player Max Projectile Velicity",
```
Comment (IceOfWraith) [inline valheimconfig.json:4541]: Duplicate DisplayName

File: valheimconfig.json [hunk 14]
```diff
@@ -94,5 +87,4644 @@
...
+    "Keywords":"Player,Projectile,Min,Velocity",
+    "FieldName":"PlayerProjectile.playerMinChargeVelocityMultiplier",
+    "InputType":"number",
+    "IsFlagArgument":false,
+    "ParamFieldName":"MonsterProjectile.playerMinChargeVelocityMultiplier",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Player Max Projectile Velicity",
```
Comment (IceOfWraith) [inline valheimconfig.json:4517]: `Velocity`

File: valheimconfig.json [hunk 15]
```diff
@@ -94,5 +87,4644 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Hud.removeDamageFlash",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "false":"false",
+      "true":"true"
+    }
+  },{
+    "DisplayName":"Remove Damage Red Flash",
```
Comment (IceOfWraith) [inline valheimconfig.json:4333]: Seems to be duplicate setting of one above with same name.

File: valheimconfig.json [hunk 16]
```diff
@@ -94,5 +87,4644 @@
...
+    "ParamFieldName":"Tameable.mortality",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{
+      "0":"Normal",
+      "1":"Essential",
+      "2":"Immortal"
+    }
+  },{
+    "DisplayName":"Tameable Settings Enable",
```
Comment (IceOfWraith) [inline valheimconfig.json:1978]: Duplicate DisplayName

File: valheimconfig.json [hunk 17]
```diff
@@ -94,5 +87,4644 @@
...
+    "FieldName":"Wagon.wagonBaseMass",
+    "InputType":"number",
+    "IsFlagArgument":false,
+    "ParamFieldName":"Wagon.wagonBaseMass",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"20",
+    "EnumValues":{},
+    "Suffix": "mass"
+  },{
+    "DisplayName":"Wagon Base Mass",
```
Comment (IceOfWraith) [inline valheimconfig.json:2049]: Duplicate DisplayName

File: valheimconfig.json [hunk 18]
```diff
@@ -94,5 +87,4644 @@
...
+    "Keywords":"Ward,Range",
+    "FieldName":"Ward.wardRange",
+    "InputType":"number",
+    "IsFlagArgument":false,
+    "ParamFieldName":"Ward.wardRange",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"20",
+    "EnumValues":{}
+  },{
+    "DisplayName":"Ward Range",
```
Comment (IceOfWraith) [inline valheimconfig.json:4718]: Duplicate DisplayName

File: valheimconfig.json [hunk 19]
```diff
@@ -94,5 +87,4644 @@
...
+    "FieldName":"StaminaUsage.bows",
+    "InputType":"number",
+    "IsFlagArgument":false,
+    "ParamFieldName":"StaminaUsage.bows",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{},
+    "Suffix": "%"
+  },{
+    "DisplayName":"clubs Stamina Usage",
```
Comment (IceOfWraith) [inline valheimconfig.json:3231]: Capital C

File: valheimconfig.json [hunk 20]
```diff
@@ -94,5 +87,4644 @@
...
+    "FieldName":"Experience.fireMagic",
+    "InputType":"number",
+    "IsFlagArgument":false,
+    "ParamFieldName":"Experience.fireMagic",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{},
+    "Suffix": "modifier"
+  },{
+    "DisplayName":"frost Magic Experience Values",
```
Comment (IceOfWraith) [inline valheimconfig.json:2399]: Capital F

File: valheimconfig.json [hunk 21]
```diff
@@ -94,5 +87,4644 @@
...
+    "IncludeInCommandLine":false,
+    "DefaultValue":"255",
+    "EnumValues":{},
+    "Suffix": "transparency"
+  },{
+    "DisplayName":"Brightness Settings Enable",
+    "Category":"Valheim Plus - Server Settings",
+    "Description":"Enabling this section gives you access to the settings up to the next section.",
+    "Keywords":"Brightness,Enable",
+    "FieldName":"GameClock.enabled",
```
Comment (IceOfWraith) [inline valheimconfig.json:3723]: Duplicate fieldname. Should be Brightness.enabled?

**Review by IceOfWraith — CHANGES_REQUESTED** [review]
First of all, let me say... Phenomenal work with this one! You should really be proud of what you accomplished by adding all these settings! I think you just took the lead for "The most lines put into a single config.json file". Lol

I've provided the replies for things that stuck out as an issue. I haven't fully tested yet until these get corrected in case they caused issues with the template. Most items are duplicated from copy/paste, or spelling errors. Nothing major in any cases. Please take a look and if I misreported something please feel free to respond to that review with a reason why I'm a dummy and mark them resolved. :)

**Comment by gkuba** [comment]
Well thanks for taking the time to look over this. Some I have already caught but I will go over this list and get them fixed and it pushed.

But yes there was a lot of copy / paste for all those sections and after the first 2k lines the text starts to blur together so again thanks for calling these out. 

The sad part is the next one to add by request is OdinsQoL and it will effectively double this.

**Comment by Greelan** [comment]
Incredible work to get all this in! I haven't looked at it in detail, but I assume at some point when you recover from doing all this that you will do a proofread? Eg in the manifest I noticed various misspellings of "Durability", and line 844 includes extra words in the FieldName (likely copied from the Keywords).

File: valheimconfig.json [hunk 22]
```diff
@@ -91,36 +91,36 @@
...
     "InputType":"checkbox",
     "IsFlagArgument":false,
     "ParamFieldName":"valheim_plus",
     "IncludeInCommandLine":false,
     "DefaultValue":"false",
     "EnumValues":{
-        "False": "false",
-        "True": "true"
+        "False": "False",
+        "True": "True"
```
Comment (IceOfWraith) [inline valheimconfig.json:118]: The right side values can be lowercase still. It's just the left side that was what AMP expects to be "True" or "False". Changing this broke the update stages. :)

File: valheimconfig.json [hunk 23]
```diff
@@ -91,36 +91,36 @@
...
     "InputType":"checkbox",
     "IsFlagArgument":false,
     "ParamFieldName":"bepinex_install",
     "IncludeInCommandLine":false,
     "DefaultValue":"false",
     "EnumValues":{
-        "False": "false",
-        "True": "true"
+        "False": "False",
+        "True": "True"
```
Comment (IceOfWraith) [inline valheimconfig.json:103]: Same here.

**Review by IceOfWraith — CHANGES_REQUESTED** [review]

File: valheimconfig.json [hunk 24]
```diff
@@ -94,5 +87,4652 @@
...
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Fire Source Ignore Private Area Check",
+    "Category":"Valheim Plus - Basic Settings",
+    "Description":"This option prevents the Fire to pull items from warded areas if it isn't placed inside of it. For convenience, we recommend this to be enabled",
+    "Keywords":"Fire,Source,ignorePrivateAreaCheck",
+    "FieldName":"FireSource.Ignore,Private,Area,Check",
```
Comment (IceOfWraith) [inline valheimconfig.json:844]: Pointed out by Greelan.

File: valheimconfig.json [hunk 25]
```diff
@@ -94,5 +87,4652 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Durability.enable",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Duriability of Armor",
```
Comment (IceOfWraith) [inline valheimconfig.json:534]: `Durability` is misspelled multiple times.

**Review by IceOfWraith — CHANGES_REQUESTED** [review]

File: valheimconfig.json [hunk 26]
```diff
@@ -94,5 +87,4652 @@
...
+    }
+  },{
+    "DisplayName":"Items Settings Enable",
+    "Category":"Valheim Plus - Basic Settings",
+    "Description":"Enabling this section gives you access to the settings up to the next section.",
+    "Keywords":"Items,Enable",
+    "FieldName":"Items.enable",
+    "InputType":"checkbox",
+    "IsFlagArgument":false,
+    "ParamFieldName":"Items.enable",
```
Comment (IceOfWraith) [inline valheimconfig.json:1439]: These should all be `...enabled` with a 'd" on the end. As it is, the settings do not update.

**Review by IceOfWraith — CHANGES_REQUESTED** [review]

File: valheimconfig.json [hunk 27]
```diff
@@ -4,51 +4,47 @@
...
+    "Category": "Valheim - General Settings",
     "ParamFieldName": "world",
     "IncludeInCommandLine": true,
     "DefaultValue": "PoweredByAMP"
-  },
-  {
+  },{
     "DisplayName": "Server Password",
-    "Description": "Password that clients must supply in order to connect",
+    "Description": "Password that clients must supply in order to connect. NOTE: Minimum password length is 5 characters & Password cant be in the server name. Default Password: 12345",
```
Comment (IceOfWraith) [inline valheimconfig.json:23]: `can't` is missing the apostrophe.

File: valheimconfig.json [hunk 28]
```diff
@@ -94,5 +87,4652 @@
...
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Enable Sleeping On Unclaimed Beds Only",
+    "Category":"Valheim Plus - Basic Settings",
+    "Description":"Enable sleeping on only unclaimed beds without setting bed as spawn. th this option enabled only beds that are not claimed by other players can be slept on without setting spawn-point using 'Shift+E'",
```
Comment (IceOfWraith) [inline valheimconfig.json:160]: `th this`

File: valheimconfig.json [hunk 29]
```diff
@@ -94,5 +87,4652 @@
...
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Kiln Stop Auto Fuel If X Coal In Chests",
+    "Category":"Valheim Plus - Basic Settings",
+    "Description":"Stops Auto Fuel looking for fuel when there is at leasts this quantity of Coal in nearby chests",
```
Comment (IceOfWraith) [inline valheimconfig.json:1619]: `least` with an extra s.

File: valheimconfig.json [hunk 30]
```diff
@@ -94,5 +87,4652 @@
...
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Duriability of Armor",
+    "Category":"Valheim Plus - Basic Settings",
+    "Description":"The Duribility of armor. The value 50 will increase the durability from 100 to 150. The value -50 will reduce the durability from 100 to 50.",
```
Comment (IceOfWraith) [inline valheimconfig.json:536]: `Durability` spelled another way again.

File: valheimconfig.json [hunk 31]
```diff
@@ -94,5 +87,4652 @@
...
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+        "False": "False",
+        "True": "True"
+    }
+  },{
+    "DisplayName":"Please Read!",
+    "Category":"Valheim Plus - Basic Settings",
+    "Description":"The settings on this page are spit up into sections. The start of a section will be labled with \"sectionName Enable\" the setting in-between are only modifiable if that section is enalbed.",
```
Comment (IceOfWraith) [inline valheimconfig.json:123]: `labeled` spelling.

Comment (IceOfWraith) [reply valheimconfig.json:123]: `enabled` spelled wrong.

File: valheimconfig.json [hunk 32]
```diff
@@ -94,5 +87,4652 @@
...
+    "IncludeInCommandLine":false,
+    "DefaultValue":"true",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Craft From Chest Ignore Private Area Check",
+    "Category":"Valheim Plus - Basic Settings",
+    "Description":"This option prevents crafting to pull items from warded areas if the player doesnt have access to it.",
```
Comment (IceOfWraith) [inline valheimconfig.json:450]: `doesn't` missing apostrophe.

File: valheimconfig.json [hunk 33]
```diff
@@ -94,5 +87,4652 @@
...
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Enable Shared Map Progression",
+    "Category":"Valheim Plus - Server Settings",
+    "Description":"With this enabled you will receive the same exploration progression as other players on the server. This will also enable the option for the server to sync everyones exploration progression on connecting to the server.",
```
Comment (IceOfWraith) [inline valheimconfig.json:4389]: `everyone's` missing apostrophe.

File: valheimconfig.json [hunk 34]
```diff
@@ -94,5 +87,4652 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Player.skipTutorials",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Disable Encumberance",
```
Comment (IceOfWraith) [inline valheimconfig.json:3023]: `encumbrance` spelled wrong.

File: valheimconfig.json [hunk 35]
```diff
@@ -94,5 +87,4652 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Map.shareMapProgression",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"Map Exporation Radius",
```
Comment (IceOfWraith) [inline valheimconfig.json:4402]: `Exploration` spelled wrong.

File: valheimconfig.json [hunk 36]
```diff
@@ -94,5 +87,4652 @@
...
+    "IsFlagArgument":false,
+    "ParamFieldName":"Player.skipIntro",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"false",
+    "EnumValues":{
+      "False":"false",
+      "True":"true"
+    }
+  },{
+    "DisplayName":"\"I Hvae Arrived\" Message",
```
Comment (IceOfWraith) [inline valheimconfig.json:2937]: `Have` spelled wrong.

File: valheimconfig.json [hunk 37]
```diff
@@ -94,5 +87,4652 @@
...
+    "ParamFieldName":"Chat.defaultWhisperDistance",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"4",
+    "EnumValues":{},
+    "Suffix": "meters"
+  },{
+    "DisplayName":"Default Normal Chat Distance",
+    "Category":"Valheim Plus - Server Settings",
+    "Description":"This determines the range in meters that you can see normal text messages by default.",
+    "Keywords":"Chat,Norlam,Distance",
```
Comment (IceOfWraith) [inline valheimconfig.json:3910]: `Normal` spelled wrong.

File: valheimconfig.json [hunk 38]
```diff
@@ -94,5 +87,4652 @@
...
+    "ParamFieldName":"Food.foodDurationMultiplier",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{},
+    "Suffix": "%"
+  },{
+    "DisplayName":"Disable Food Degradation Over Time",
+    "Category":"Valheim Plus - Player Settings",
+    "Description":"This option prevents food degrading over time - in other words, it retains its maximum benefit until it runs out instead of reducing its effect over time.",
+    "Keywords":"Food,Degredation,Over,Time",
```
Comment (IceOfWraith) [inline valheimconfig.json:2686]: `Degradation` spelled wrong.

File: valheimconfig.json [hunk 39]
```diff
@@ -94,5 +87,4652 @@
...
+    "ParamFieldName":"Stamina.dodgeStaminaUsage",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{},
+    "Suffix": "%"
+  },{
+    "DisplayName":"Encumbered Stamina Drain",
+    "Category":"Valheim Plus - Player Settings",
+    "Description":"Changes the stamina drain of being overweight by %",
+    "Keywords":"Stamina,Encombered,Drain",
```
Comment (IceOfWraith) [inline valheimconfig.json:3112]: `encumbered` spelled wrong.

File: valheimconfig.json [hunk 40]
```diff
@@ -94,5 +87,4652 @@
...
+    "ParamFieldName":"Stamina.staminaRegen",
+    "IncludeInCommandLine":false,
+    "DefaultValue":"0",
+    "EnumValues":{},
+    "Suffix": "%"
+  },{
+    "DisplayName":"Stamina Regen Delay",
+    "Category":"Valheim Plus - Player Settings",
+    "Description":"Changes the delay until stamina regeneration sets in by %",
+    "Keywords":"Stamina,Regen,Dealy",
```
Comment (IceOfWraith) [inline valheimconfig.json:3177]: `Delay` spelled wrong.

**Review by IceOfWraith — CHANGES_REQUESTED** [review]
Okay, I think I caught all the spelling issues that Greelan was referring to. Please, be sure to do a find/replace on all because I only marked one instance of each misspelling. There were more for some of them.

Honestly, less mistakes than I'd expect with 4,000 lines! I know with Space Engineers I had quite a few to revise. Lol

**Comment by gkuba** [comment]
Ok I think I got all that. Thanks you guys for looking over this you can clearly see the days I wasn't feeling well at all but couldn't lay down any longer. Sorry about the mess. 

I double checked there shouldn't be any more mixed up FieldName and Keywords fields.

**Review by IceOfWraith — APPROVED** [review]
I've tested the basic functionality and spot checked settings. It seems to be working as expected. Very nicely done!

**Pull request merged**